src/shallow/Snapshot/Serialize/Snapshot.lua

```diff
@@ -0,0 +1,133 @@
+local RoactRoot = script.Parent.Parent.Parent.Parent
+
+local Markers = require(script.Parent.Markers)
+local ElementKind = require(RoactRoot.ElementKind)
+local Type = require(RoactRoot.Type)
+local Ref = require(RoactRoot.PropMarkers.Ref)
+
+local function sortSerializedChildren(childA, childB)
+	return childA.hostKey < childB.hostKey
+end
+
+local Snapshot = {}
+
+function Snapshot.type(wrapperComponent)
+	local kind = ElementKind.fromComponent(wrapperComponent)
+
+	local typeData = {
+		kind = kind,
```

Comment:
I made a longer comment above, but it's useful for function components, because I can't serialize their function. It's only going to show up as a AnonymousFunction symbol